sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertIntoHiveTableSuite.scala

```diff
@@ -216,6 +215,33 @@ class InsertIntoHiveTableSuite extends QueryTest with TestHiveSingleton with Bef
     sql("DROP TABLE hiveTableWithStructValue")
   }
 
+  test("SPARK-10216: Avoid empty files during overwrite into Hive table with group by query") {
+    val testDataset = hiveContext.sparkContext.parallelize(
+      (1 to 2).map(i => TestData(i, i.toString))).toDF()
+    testDataset.registerTempTable("testDataset")
+
+    val tmpDir = Utils.createTempDir()
+    sql(
+      s"""
+        |CREATE TABLE table1(key int,value string)
+        |location '${tmpDir.toURI.toString}'
+      """.stripMargin)
+    sql(
+      """
+        |INSERT OVERWRITE TABLE table1
+        |SELECT count(key), value FROM testDataset GROUP BY value
```

Comment:
Seems you want to explicitly control the number of shuffle partitions? Otherwise, this test will not testing anything if the number of shuffle partitions is set to 2 by any chance?